ui/src/ReactCustomHooks/useWebSocket/README.md

```diff
@@ -0,0 +1,62 @@
+# WebSocketHook
+
+This hook in an abstraction of the [WebSocket API](https://developer.mozilla.org/en-US/docs/Web/API/WebSocket).
+
+This hook will manage the opening and closing of the websocket when the 
+component using this hook mounts and unmounts. It also will manage the 
+websocket through it's lifecycle, and will call through to provided 
+callbacks as events occur.
+
+It will return a function which will allow the sending of data (when 
+the websocket allows), and the current status if the websocket.
+
+Details on the usage can be found below. Finally, a set of websocket 
+states are also exported alongside the hook, under the name `WEBSOCKET`.
+
+## Usage
+
+Create a new websocket:
+
+```
+const { send, currentState } = useWebSocket(
+  func() => WebSocket,
```

Comment:
`func() => WebSocket | url`

Reply:
Change in API since we last discussed it - expects a callback to return a websocket. URL is one of many options which could be provided - felt more flexible/intuitive to have one expected input here